Beta_Integrations/integration-Vertica.yml

```diff
@@ -0,0 +1,175 @@
+    DATABASE = demisto.params().get('database')
+    PORT = int(demisto.params().get('port', 5433))
+    SERVER = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
+
+    ''' HELPER FUNCTIONS '''
+
+    def check_if_datetime(v):
+        """
+        Parses datetime object into string
+        """
+        if isinstance(v, datetime.datetime):
+            return v.strftime("%Y-%m-%dT%H:%M:%S")
+        return v
+
+
+    def connect_db():
+        db_params = {
+            'host': SERVER,
+            'port': PORT,
+            'user': USERNAME,
+            'password': PASSWORD,
+            'database': DATABASE,
+            'connection_timeout': 5
+        }
+        try:
+            connection = vertica_python.connect(**db_params)
+        except vertica_python.errors.ConnectionError:
+            return_error('Could not connect to DB, please re-check DB params.')
+        return connection
+
+    CONNECTION = connect_db()
+    CURSOR = CONNECTION.cursor('dict')
+
+    ''' COMMANDS + QUERY FUNCTIONS '''
+
+    def test_module():
+        """
+        Performs basic query on default system tables
+        """
+        CURSOR.execute("SELECT * FROM system_tables ORDER BY table_schema, table_name LIMIT 2;")
+        rows = CURSOR.fetchall()
+        if CURSOR.rowcount == 0:
+            return_error('No results were returned from the DB.')
+
+
+    def query_command():
+        """
+        Execute a query against the DB
+        """
+        # Init main vars
+        contents = []
+        context = {}
+        title = 'Vertica Query'
+        # Get arguments from user
+        query = demisto.args().get('query')
+        limit = int(demisto.args().get('limit', 50))
+        # Query and get raw response (list of ordered dicts)
+        rows = query_request(query)
+        # Parse response into context & content entries
+        if rows:
+            if limit:
+                rows = rows[:limit]
+            title = 'Vertica Query Results:'
+
+            for i, row in enumerate(rows):
+                rows[i] = { underscoreToCamelCase(k): check_if_datetime(v) for k, v in row.items() }
+
+            contents = rows
+            context['Vertica.Query(val.Query && val.Query === obj.Query)'] = {
+                'Query': query,
+                'Rows': rows
+            }
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': contents,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(title, contents, removeNull=True),
+            'EntryContext': context
+        })
+
+
+    def query_request(query):
+        try:
+            CURSOR.execute(query)
+        except vertica_python.errors.MissingRelation:
+            return_error('Error while executing query.')
```

Comment:
Maybe add the real error as well?

Reply:
tried, it has issues accessing the message itself and throws exception in exception, so I kept it like that for now